### PR TITLE
[bitnami/rabbitmq] Update readinessProbe to avoid deadlocks

### PR DESCRIPTION
**Pull request by javsalgar**
**Description of the change**

Change readiness probe command as it was causing issues

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes https://github.com/bitnami/charts/issues/3783

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by mltsy** [comment]
This looks right to me, although I have not tested this solution (so this is just from knowing the issue and looking at the documentation)

**Review by dani8art — APPROVED** [review]
LGTM, please resolve conflict in `Chart.yaml`

**Comment by mltsy** [comment]
One thing that concerns me about this is what happens in the case of a rolling upgrade of RabbitMQ version.  I guess I don't even know if the chart supports rolling upgrade, or what it does if you try to upgrade the rabbit version!  So I will open a new issue for this.  However, in any case, this setting would *not* support a rolling upgrade with no downtime, since it doesn't wait for the sync to complete before moving on to the next node (which you would want in the case of a rolling upgrade - I think there may be some features needed around specifying an upgrade strategy, that might trigger different rabbitmq-specific actions/checks during the upgrade process)

**Comment by mltsy** [comment]
Update: it looks like a rolling upgrade is the default for this chart, which I *think* means this PR changes the default from one that is safe (waits for mirrors to sync) to one that is unsafe, in the case of durable mirrored queues.  Right?

Example: If you have a classic mirrored queue, and perform a `helm upgrade` it will start a rolling upgrade that doesn't wait for mirrors to re-sync after restarted, and therefore could cause downtime when it finally shuts down the master node for the mirrored queue (assuming none of the mirrors are synced yet) - and could actually result in data-loss if `ha-promote-on-shutdown` is set to `always`.

I think there needs to be a new parameter for this called something like `upgradeOptions.waitForSync` that can be true or false, so that if you're spinning up a cluster.

Although, now that I'm thinking about it, the way it's supposed to work is that the first node to start up is the last node to shut down (and therefore the master node, which doesn't wait for other pods in order to start working).  In the case of this chart, that's always supposed to be rabbitmq-0.  However, in the case of a total cluster failure or something, the chart doesn't know which node was the master.  So I guess the above suggestion (`upgradeOptions.waitForSync`) might still be a useful option, but it should default to true for regular upgrade situations, rather than false as this PR does.

**Comment by dani8art** [comment]
I see your point, so you mean to have two types of probes, one will wait for mirrors and the other one will just ping the node to see if it is working, and then a new property `upgradeOptions.waitForSync` that will decide which one to use.

On the other hand, I think the best default would `true` as you have argued.

Thanks!!

File: bitnami/rabbitmq/templates/statefulset.yaml [hunk 1]
```diff
@@ -228,7 +228,7 @@ spec:
               command:
                 - /bin/bash
                 - -ec
-                - rabbitmq-diagnostics -q check_running
+                - rabbitmq-diagnostics -q ping
```
Comment (dani8art) [inline bitnami/rabbitmq/templates/statefulset.yaml:231]: Add wait for sync options

**Review by dani8art — COMMENTED** [review]
Please, let's try to integrate the latest suggestions

**Comment by mltsy** [comment]
Wow!  Actually, this issue addresses the syncing issue in a much better way, I think: https://github.com/bitnami/charts/issues/3931

Which means, this PR is probably fine.  The livenessProbe shouldn't have to worry about any kind of syncing, as long as the sync-wait is handled on pod shutdown (where it should really be handled anyway).  Right?  As long as the pod is allowed the time it needs to shutdown!

**Comment by dani8art** [comment]
Waiting for this https://github.com/bitnami/bitnami-docker-rabbitmq/pull/142 to be solved

**Pull request merged**